src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -65,14 +65,15 @@ internal partial class VisualStudioDiagnosticAnalyzerService : IVisualStudioDiag
             _globalOptions = globalOptions;
         }
 
-        public void Initialize(IServiceProvider serviceProvider)
+        public async Task InitializeAsync(IAsyncServiceProvider serviceProvider, CancellationToken cancellationToken)
         {
-            _serviceProvider = serviceProvider;
+            _serviceProvider = (IServiceProvider)serviceProvider;
 
             // Hook up the "Run Code Analysis" menu command for CPS based managed projects.
-            var menuCommandService = (IMenuCommandService)_serviceProvider.GetService(typeof(IMenuCommandService));
+            var menuCommandService = await serviceProvider.GetServiceAsync<IMenuCommandService, IMenuCommandService>().ConfigureAwait(false);
```

Comment:
⚠️ Make sure this is not using the `GetServiceAsync` method provided by the VS SDK. If it is, I strongly suggest adding that extension method to the banned API file to avoid accidental future use.

Reply:
I'll send a separate PR to ban these